inc/classes/class-abstract-render.php

```diff
@@ -0,0 +1,71 @@
+<?php
+namespace WP_Rocket;
+
+use WP_Rocket\Interfaces\Render_Interface;
+
+/**
+ * Handle rendering of HTML content created by WP Rocket.
+ *
+ * @since 3.0
+ * @author Remy Perona
+ */
+abstract class Abstract_Render implements Render_Interface {
+	/**
+	 * Path to the templates
+	 *
+	 * @since 3.0
+	 * @author Remy Perona
+	 *
+	 * @var string
+	 */
+	private $template_path;
+
+	/**
+	 * Constructor
+	 *
+	 * @since 3.0
+	 * @author Remy Perona
+	 *
+	 * @param string $template_path Path to the templates.
+	 */
+	public function __construct( $template_path ) {
+		$this->template_path = $template_path;
+	}
+
+	/**
+	 * Renders the given template if it's readable.
+	 *
+	 * @since 3.0
+	 * @author Remy Perona
+	 *
+	 * @param string $template Template slug.
+	 * @param array  $data     Data to pass to the template.
+	 */
+	public function generate( $template, $data = array() ) {
+		$template_path = $this->get_template_path( $template );
+
+		if ( ! rocket_direct_filesystem()->is_readable( $template_path ) ) {
+			error_log( $template_path );
```

Comment:
I know about this one, will remove it later :)

Reply:
Lol ^^